crates/cdk/src/wallet/multi_mint_wallet.rs

```diff
@@ -0,0 +1,304 @@
+//! MultiMint Wallet
+//!
+//! Wrapper around core [`Wallet`] that enables the use of multiple mint unit pairs
+
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use serde::{Deserialize, Serialize};
+use tokio::sync::Mutex;
+use tracing::instrument;
+
+use super::Error;
+use crate::amount::SplitTarget;
+use crate::nuts::{CurrencyUnit, SecretKey, SpendingConditions, Token};
+use crate::types::{Melted, MintQuote};
+use crate::{Amount, UncheckedUrl, Wallet};
+
+#[derive(Debug, Clone)]
+pub struct MultiMintWallet {
+    pub wallets: Arc<Mutex<HashMap<WalletKey, Arc<Wallet>>>>,
```

Comment:
I don't think `Wallet` needs to be wrapped in an `Arc` since it is already `Clone`.